xarray/core/dataset.py

```diff
@@ -7074,5 +7074,43 @@ def query(
         # apply the selection
         return self.isel(indexers, missing_dims=missing_dims)
 
+    def drop_duplicates(
+        self,
+        dims: Union[None, Hashable, Sequence[Hashable]] = None,
```

Comment:
We currently allow implicit `None` - you can do either of those:

```suggestion
        dims: Union[Hashable, Iterable[Hashable]] = None,
```

```suggestion
        dims: Optional[Union[Hashable, Iterable[Hashable]]] = None,
```